lib/AST/Decl.cpp

```diff
@@ -1056,6 +1056,8 @@ ImportDecl *ImportDecl::create(ASTContext &Ctx, DeclContext *DC,
   auto D = new (ptr) ImportDecl(DC, ImportLoc, Kind, KindLoc, Path);
   if (ClangN)
     D->setClangNode(ClangN);
+  auto realName = Ctx.getRealModuleName(Path.front().Item);
+  D->setRealModuleName(realName);
```

Comment:
Also, do you want to do something special if the real and alias names are identical so that `RealModuleName` will just be empty? That would make the common case avoid using the scratch buffer.

(Or will `ASTContext::getRealModuleName()` return an empty identifier so that a special case here is not needed? I've lost track of this aspect of the implementation.)

Reply:
Will pass in a lookup option so it's empty for non-aliasing cases.